### PR TITLE
MLP Hyperparameter Optimization Binary Clasification Model

### PR DESCRIPTION
**Pull request by MikeLippincott**
This PR contains code that does few things:

Does Hyperparameter optimization on a binary classification MLP where the goal is to predict if a cell was either treated with pyroptotic inducing agents or not based on features.

From the optimized hyperparameters, create the MLP, train the MLP, and test the MLP.

note: I opted against placing functions in this notebook in a helper file as these functions define the workflow and the model architecture. 

Further optimization can be performed, I am thinking about narrowing the search space of the optimizer.

Future steps will be designing a multi-class MLP to predict pyroptosis and apoptosis given that there are other treatments other than LPS and other dosages.

### CODE REVIEW

**Comment by axiomcura** [comment]
Very neat Notebook! However, one general comment. 

You have created lots of functions in your analysis, but they are quite large! I would suggest adding some doc-string documentation to your function to make it easier to read what you're doing! 

Overall good PR!

**Review by d33bs — CHANGES_REQUESTED** [review]
Nice work! I have comments here and also from ReviewNB which are forthcoming (will post after the Github comments). Please don't hesitate to let me know if you have any questions. 

I'd request changes on removing the `pyc` files from this PR before approval. All other comments and suggestions are flexible to what you think is best.

**Comment by MikeLippincott** [comment]
<p>Documentation added, this is a global variable for use in later functions (optimization function and model training function). </p><p> </p><p>An alternative is to make each of these functions methods in a class and have constants in the init method so that it is accessible. Thoughts?</p><p> </p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470602909'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p>I had not thought of this, thank you. Changes have been made, easy change.</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470499400'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p>Constructed class docstring, thank you for pointing this out. Documentation of functions, and classes is something that I need to work on throughout this file.</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470602983'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p>At some point during testing yes, but I don't think this needs to be present anymore. I am removing it.</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470499445'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p>I am seeing a theme of improving documentation. This is something that I am taking into account for future code. Thank you :)</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470603033'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p>Documentation added.</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470603033'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p>added spelling fix!</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470603064'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p>Added a docstring</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470603126'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p> I think so, this is something I will be working on in a later PR. I want to place functions in a class and reuse functions. I think this is a big refactor for this PR though.</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470603154'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p>Added Documentation</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470499466'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by MikeLippincott** [comment]
<p>doc string added</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470603218'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: 1.Machine_Learning_MLP/Hyperparameter_Optimization.ipynb
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 1.Machine_Learning_MLP/Hyperparameter_Optimization.ipynb:1]: There was supposed to be but my per-commit was raising issues and would not allow me to push the converted file so I removed it now that we have ReviewNB

**Comment by d33bs** [comment]
<p>Sounds good!</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470603261'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by d33bs** [comment]
<p>👍</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470603188'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by d33bs** [comment]
<p>I understand, sounds good!</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470603154'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by d33bs** [comment]
<p>👍</p>

---

 <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/4/discussion/#comment-1470602983'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: 1.Machine_Learning_MLP/Hyperparameter_Optimization.ipynb
[the remark this answers is not in the document]
Comment (d33bs) [reply 1.Machine_Learning_MLP/Hyperparameter_Optimization.ipynb:1]: Sounds good!

**Review by d33bs — APPROVED** [review]

**Pull request merged**